### DISCUSSION TITLE
How to specify a DNS server or customize a DOH address

### DISCUSSION THREAD

**Discussion by sln162** [MicrosoftEdge/WebView2Feedback#4088] [Q&A]
Is there any parameter that can specify the DNS server for WebView2 or a custom address for DOH? Edge can change the custom DOH server address through settings, and how does WV2 achieve this?

Because some of our users have unstable dns for the system or operators, we hope to customize DNS servers, even for DOH server addresses.

**Comment by ghost** [comment]
I have the same question too. I want to customize a DOH server. 
For a dns, you may try command line switches.

**Reply by sln162** [reply]
Modifying user DNS is not a good behavior, so we finally used DnsOverHttpsMode and DnsOverHttpsTemplates to customize the DOH address of WV2

**Comment by ghost** [comment]
So can we provide a page like edge://settings for the users?
________________________________
发件人: sln162 ***@***.***>
发送时间: 2024年10月1日 13:46
收件人: MicrosoftEdge/WebView2Feedback ***@***.***>
抄送: loueisun ***@***.***>; Comment ***@***.***>
主题: Re: [MicrosoftEdge/WebView2Feedback] How to specify a DNS server or customize a DOH address (Discussion #4088)


Modifying user DNS is not a good behavior, so we finally used DnsOverHttpsMode and DnsOverHttpsTemplates to customize the DOH address of WV2

―

**Reply by sln162** [reply]
https://github.com/MicrosoftEdge/WebView2Feedback/issues/4089

**Comment by ghost** [comment]
Is it possible to set it without administration account?
Access denied in regedit editing on my computer.


________________________________
发件人: sln162 ***@***.***>
发送时间: 2024年10月1日 13:46
收件人: MicrosoftEdge/WebView2Feedback ***@***.***>
抄送: loueisun ***@***.***>; Comment ***@***.***>
主题: Re: [MicrosoftEdge/WebView2Feedback] How to specify a DNS server or customize a DOH address (Discussion #4088)


Modifying user DNS is not a good behavior, so we finally used DnsOverHttpsMode and DnsOverHttpsTemplates to customize the DOH address of WV2

―

**Reply by sln162** [reply]
Yes, without administrator privileges, it will be impossible to change the registry
